controller/src/main/java/io/pravega/controller/store/stream/ZKStreamMetadataStore.java

```diff
@@ -460,6 +463,35 @@ public void unregisterBucketListener(int bucket) {
                           });
     }
 
+    /**
+     * When managing the Controller's metadata in Zookeeper, we explicitly create parent bucket zNodes (so they are of
+     * type "zNode"). Otherwise, they may be inadvertently created as Zookeeper "containers" by Curator. This would lead
```

Comment:
Can you point in this comment to curator documentation that says this, please?

Reply:
Added a reference for this. Actually, the first line of the page says: `NOTE: Most Curator recipes will autocreate parent nodes of paths given to the recipe as CreateMode.CONTAINER`.